compiler/fir/raw-fir/fir-common/src/org/jetbrains/kotlin/fir/builder/Context.kt

```diff
@@ -18,7 +18,8 @@ import org.jetbrains.kotlin.name.Name
 class Context<T> {
     lateinit var packageFqName: FqName
     var className: FqName = FqName.ROOT
-    val currentClassId get() = ClassId(packageFqName, className, firFunctionTargets.isNotEmpty())
+    val localBits: MutableList<Boolean> = mutableListOf()
```

Comment:
You could make this `internal` since it is not used outside the package. (In fact, maybe you could even copy `withChildClassName` in here to make `localBits` private, and then leave `withChildClassName` in `BaseFirBuilder` just to delegate it to `Context.withChildClassName`.)

Reply:
It's reasonable to make it `internal`, but `withChildClassName`, which is `inline`, requires to access only public items. As mentioned at the other comment, I prefer the current data model of `Context`, without any default-ish, guideline logic to manipulate parts of context. Actually, such class-related utils are about creating (raw) `fir` elements (in particular, class), so I think those should belong to the builder(s), not context.